src/ovirtapi/index.js

```diff
@@ -203,7 +207,7 @@ const OvirtApi = {
       contentType: 'application/xml',
     })
   },
-  addNewSnapshot ({ vmId, snapshot }: { vmId: string, snapshot: Object }): Promise<Object> {
+  addNewSnapshot ({ vmId, snapshot }: { vmId: string, snapshot: SnapshotType }): Promise<Object> {
     assertLogin({ methodName: 'addNewSnapshot' })
     const input = JSON.stringify(OvirtApi.internalSnapshotToOvirt({ snapshot }))
```

Comment:
Looking at the REST docs, looks like you only need the description to create a snapshot.  Using a full `SnapshotType` here is a bit much and requires the transform default values to prevent errors with undefined values.

If you need to support fine grained details (`persist_memorystate` or `disk_attachments`) making those parameters helps explicitly define how the API function works.  In other words, don't require a developer to know exactly how the rest call works when they look at it.

Reply:
Yeah, you right, here must be `ApiSnapshotType`.

Reply:
inputs are internal types (or at least partials) while the Promise returned would be the Api type (or an error).

eventually it would be nice to have the Apis typed properly, although keeping it updated by hand is less then ideal